### ISSUE TITLE
setcontent is much slower in Bolt 5.0 compared to Bolt 3.7

### ISSUE THREAD

**Issue by sjoert-skrepr** [bolt/core#2848]
I'm currently upgrading a fairly large Bolt 3.7 website to Bolt 5.0 which both use a MariaDB 10.3.31 database. After completing the homepage, I noticed the loading times were much higher compared to the same page in Bolt 3.7. In Bolt 5.0 this page runs 178 queries in ~720ms, while 3.7 does 115 queries in ~40ms. Looking at the profiler, I seems this query is the culprit:

```
SELECT b0_.id             AS id_0,
       b0_.content_type   AS content_type_1,
       b0_.status         AS status_2,
       b0_.created_at     AS created_at_3,
       b0_.modified_at    AS modified_at_4,
       b0_.published_at   AS published_at_5,
       b0_.depublished_at AS depublished_at_6,
       b0_.author_id      AS author_id_7
FROM bolt_content b0_
         LEFT JOIN bolt_field b1_ ON b0_.id = b1_.content_id AND b1_.type IN
                                                                 ('generic', 'markdown', 'data', 'hidden', 'file',
                                                                  'templateselect', 'set', 'number', 'filelist',
                                                                  'image', 'slug', 'email', 'collection', 'text',
                                                                  'embed', 'html', 'date', 'checkbox', 'select',
                                                                  'imagelist', 'block', 'textarea', 'article',
                                                                  'redactor')
         LEFT JOIN bolt_field_translation b2_ ON b1_.id = b2_.translatable_id
WHERE b0_.content_type = 'blocks'
  AND ((JSON_EXTRACT(b2_.value, '$[0]') = 'newsletter' AND b1_.parent_id IS NULL AND b1_.name = 'slug') AND
       b0_.status = 'published')
ORDER BY b0_.published_at DESC
LIMIT 1;
```

It did about 15 of these, each taking between 35ms and 40ms. So that adds up. They most likely get invoked when there is a `{% setcontent %}` in a twig template. I played around with it to see if I could make it faster. Removing the `ORDER BY` clause reduces the query time to 1ms. I tried setting an index on `published_at` but it didn't help.

Here is the output of `EXPLAIN`:

```
| id | select_type | table | type   | possible_keys                                             | key     | key_len | ref                          | rows | Extra                                        |
|----|-------------|-------|--------|-----------------------------------------------------------|---------|---------|------------------------------|------|----------------------------------------------|
|  1 | SIMPLE      | b2_   | ALL    | field_translation_unique_translation,IDX_5C60C0542C2AC5D3 | NULL    | NULL    | NULL                         | 5916 | Using where; Using temporary; Using filesort |
|  1 | SIMPLE      | b1_   | eq_ref | PRIMARY,IDX_4A2EBBE584A0A3ED,IDX_4A2EBBE5727ACA70         | PRIMARY | 4       | webs_new.b2_.translatable_id | 1    | Using where                                  |
|  1 | SIMPLE      | b0_   | eq_ref | PRIMARY,content_type_idx,status_idx                       | PRIMARY | 4       | webs_new.b1_.content_id      | 1    | Using where                                  |
```

**Comment by bobdenotter** [comment]
@sjoert-skrepr I've moved the issue to the correct repo: `/bolt/core`! :-)

**Comment by sjoert-skrepr** [comment]
@bobdenotter I think I found the issue. I noticed doing a query as simple as this one was already quite slow (35ms in my case):
```
SELECT *
FROM bolt_field_translation b2_
WHERE JSON_EXTRACT(b2_.value, '$[0]') = 'nieuwsbrief';
```
My understand here is, because the `JSON_EXTRACT` is in the `WHERE` clause, mysql will have to do a JSON parse on every row in the table. In my case there are 7214 rows. Another downside of this is that indices can't be used even if one is present. Changing the query to this one brought the query time down to 5ms:
```
SELECT *
FROM bolt_field_translation b2_
WHERE b2_.value = '["nieuwsbrief"]';
```
After that adding an index to `bolt_field_translation.value` (used a length of 64) brought the query time down to less than 1ms. Making the same changes to the query from the original post yielded the same results.

I understand that this won't work when the JSON structure in the `value` column is more elaborate (like an image field type). But there are a lot of field types where it is just a single value, like slug, text, checkbox etc.
